src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -1953,6 +1955,59 @@ void QgsSymbolLayerUtils::externalGraphicToSld( QDomDocument &doc, QDomElement &
   }
 }
 
+void QgsSymbolLayerUtils::parametricSvgToSld( QDomDocument &doc, QDomElement &graphicElem,
+    const QString& path, const QColor& color, double size, const QColor& outlineColor, double outlineWidth )
+{
+  // Parametric SVG paths are an extension that few systems will understand, but se:Graphic allows for fallback
+  // symbols, this encodes the full parametric path first, the pure shape second, and a mark with the right colors as
+  // a last resort for systems that cannot do SVG at all
+
+  // encode parametric version with all coloring details (size is going to be encoded by the last fallback)
+  graphicElem.appendChild( doc.createComment( "Parametric SVG" ) );
+  QString parametricPath = getSvgParametricPath( path, color, outlineColor, outlineWidth );
+  QgsSymbolLayerUtils::externalGraphicToSld( doc, graphicElem, parametricPath, QStringLiteral( "image/svg+xml" ), color, -1 );
+  // also encode a fallback version without parameters, in case a renderer gets confused by the parameters
+  graphicElem.appendChild( doc.createComment( "Plain SVG fallback, no parameters" ) );
+  QgsSymbolLayerUtils::externalGraphicToSld( doc, graphicElem, path, QStringLiteral( "image/svg+xml" ), color, -1 );
+  // finally encode a simple mark with the right colors/outlines for renderers that cannot do SVG at all
+  graphicElem.appendChild( doc.createComment( "Well known marker fallback" ) );
+  QgsSymbolLayerUtils::wellKnownMarkerToSld( doc, graphicElem, "square", color, outlineColor, Qt::PenStyle::SolidLine, outlineWidth, -1 );
+
+  // size is encoded here, it's part of se:Graphic, not attached to the single symbol
+  if ( size >= 0 )
+  {
+    QDomElement sizeElem = doc.createElement( "se:Size" );
+    sizeElem.appendChild( doc.createTextNode( qgsDoubleToString( size ) ) );
+    graphicElem.appendChild( sizeElem );
+  }
+}
+
+
+QString QgsSymbolLayerUtils::getSvgParametricPath( const QString& basePath, const QColor& color, const QColor& borderColor, double borderWidth )
+{
+  QUrl url = QUrl();
+  if ( color.isValid() )
+  {
+    url.addQueryItem( "fill", color.name() );
+    url.addQueryItem( "fill-opacity", encodeSldAlpha( color.alpha() ) );
+  }
+  if ( borderColor.isValid() && borderWidth > 0 )
+  {
+    url.addQueryItem( "outline", borderColor.name() );
+    url.addQueryItem( "outline-opacity", encodeSldAlpha( borderColor.alpha() ) );
+    url.addQueryItem( "outline-width", QString::number( borderWidth ) );
```

Comment:
It's possible that an svg could use the outline-width parameter without having an outline-color. Actually that applies to all parameters, eg fill opacity without paramaterised fill colors.